src/rgw/driver/rados/rgw_pubsub_push.cc

```diff
@@ -222,17 +227,24 @@ class RGWPubSubAMQPEndpoint : public RGWPubSubEndpoint {
             w.complete(boost::system::error_code{}, rc);
           }
         });
-        return w.async_wait(yield);
+        const auto rc = w.async_wait(yield);
```

Comment:
why dont upu check `rc < 0` here, and increment the failure counter accordingly?
if `amqp::publish_with_confirm()` fails in line 222, this should be reflected here